.gitlab/common/macos.yml

```
@@ -16,6 +16,24 @@
       dda inv -- -e macos.remove-inactive-versions -l python -t "$PYTHON_VERSION" || true
       dda inv -- -e macos.remove-inactive-versions -l go -t "$(cat .go-version)" || true
     fi
+  # Create custom temporary folder to isolate jobs from each other
+  # We have to symlink it to /tmp/gitlabci to avoid some path length issues (sockets should be <= 104 characters on MacOS)
+  - |
+    if [ -z "$TMPDIR" ]; then
+      echo "TMPDIR must be set" >& 2
+      exit 1
+    fi
+    # Remove the trailing slashes etc.
+    export OLDTMPDIR="$(readlink -m "$TMPDIR")"
+    sudo rm -rf "$OLDTMPDIR/gitlabci"
+    NEWTMPDIR="$OLDTMPDIR/gitlabci/$CI_JOB_ID"
+    mkdir -p "$NEWTMPDIR"
+    sudo chmod 777 "$NEWTMPDIR"
+    sudo chown "$USER":staff "$NEWTMPDIR"
+    sudo unlink /tmp/gitlabci 2> /dev/null || true
+    sudo ln -s "$NEWTMPDIR" /tmp/gitlabci
+    export TMPDIR="/tmp/gitlabci/"
```

```suggestion
    export TMPDIR="/tmp/gitlabci"
```
(the semantics of the directory is conveyed by the variable name, plus derived paths such as below `$TMPDIR/datadog-agent-build/bin` would expand to `/tmp/gitlabci//datadog-agent-build/bin`)

---

This ends with a slash by default, removed slash to see if jobs are still properly behaving